SU-HSDO/suhumsci
# STN-558 : Add Hero Animations

# READY FOR REVIEW

## Summary
Implement hero animations for both Colorful and Traditional. The Colorful themed hero animation will fade in the image, then the text. The Traditional themed hero animation will slide down and fade in the text.

Hero animations are an animation enhancement. They will only display is the animation enhancement theme toggle has been activated.

Hero animations will only render when the user scrolls down the page the first time. They will not render more than once and they will not render when scrolling up the page.

## Need Review By (Date)
asap

## Urgency
high

## Steps to Test
1. In your CLI, run `npm run test` and confirm all tests pass.
2. Set the theme setting to Colorful and turn on the animation enhancement theme setting
3.  Go to http://sparkbox-sandbox.suhumsci.loc/hero-height-refinements
- [x] Confirm the hero animation for Colorful fades in the image, and then the text.
- [x] Confirm the hero animation triggers upon scroll when the hero appears in the browser viewport
- [x] Confirm the hero animation once renders once as your scroll down the page. Upon scroll back up the page, the hero animations should not render again.
4. Because the hero and the carousel share many of the same classes, go to http://sparkbox-sandbox.suhumsci.loc/qa/carousel-hero-image and confirm the carousel does not animate.
5. Turn off the animation enhancement theme setting and return to http://sparkbox-sandbox.suhumsci.loc/hero-height-refinements
 -  Confirm the hero animation do not render when the hero animation setting is unchecked
6. Repeat the steps above for the Traditional theme noting that the animation for Traditional should slide down and fade in. The overlay also also a very subtle transition from 90 to 100% opacity.
